docs/UserGuide.md

```diff
@@ -267,7 +267,20 @@ Format: `exit`
 
 ### Saving the data
 
-Vax@NUS saves your current date in the hard disk automatically after any command. There is no need to save manually.
+Vax@NUS saves your current data into your computer automatically after any command. There is no need to save manually.
+
+:information_source: **NOTE:**  VAX@NUS will display our sample data file if no data file is found from your computer.
+
+### Editing the data
+
+VAX@NUS data is saved as a JSON file [JAR file location]/data/studentbook.json. Advanced users are welcome to update data directly by editing that data file.
+
+Please refer to the [Input Formats](#input-formats) section when editing the data file to conform to the required format .
+
+> :warning: **If your changes to the data file makes it an invalid format, VAX@NUS will discard all data and start with an empty data file at the next run**: Be very careful!
+
+:information_source: **NOTE:** VAX@NUS will load an empty data file if you have cleared your data with the `clear` command right before exiting VAX@NUS.
+Additionally, VAX@NUS is unable to detect changes in the matriculation number if it is edited directly from our data file.
```

Comment:
Maybe this line could also clarify the reason why it is unable to detect changes is because matriculation number is not supposed to be editable?

Reply:
Hmmm.. I don't think its needed since we are just letting the user knows that even though they can't edit matric number through the edit command... They can do so on the JSON file but our app can't detect any changes in matric number... So if they tried to file a bug against not being able to edit matric number through edit command yet they can edit through the JSON file then we can refer them to this line written in the UG.

Reply:
Ohhh I understood 'unable to detect changes' as if matric number is edited in json file, the application will not detect the edit! Not sure if it is just me but would this phrasing be ambiguous?

Reply:
I don't quite understand this sentence. What does it mean to 'detect'? Does it mean to reflect the changes in the UI?

Reply:
Yup, I meant to say that the UI will still reflect the new matric number if you edit the matric number field directly from the JSON file. Would "VAX@NUS will reflect the new matriculation number if it is edited directly from the JSON file. " be clearer?

Reply:
Wouldn't our app reflect any changes to any field in the json file? Would mentioning only matric number gives a false impression about the difference between editing matric number and other fields in the json file?

Reply:
Hmm... I understand your concern...  But I think the problem stems from the unmodifiable matric number field. 

The JSON file allows every field to be edited but the edit command disallow the matric number to be edited.. 

I'm not really sure how the user will think if we specified that matriculation number cannot be edited in the edit command section... If they read and think that matric number is unmodifiable throughout the app... Then they might think that there is a functionality bug when it comes to the JSON file since it allows the user to edit the matric number....

Reply:
Ah I forgot about the app making matric number unmodifiable. 

What if we say that, while users can edit the matric number in JSON file even though the GUI doesn't allow, extra care must be taken to ensure data integrity. Specifically, if the user decides to edit the matric number of a student, they must either (1) edit any appointments for this student to reference the new matric number, or (2) delete all appointments linked to the student. 

Btw what would happen if I edit matric number in JSON and the matric number in appointments are not edited? We should specify the expected behaviour in this scenario somewhere in UG also?

Reply:
Would writing it down to inform the users that they can edit the matric number in JSON contradict with our edit command behavior? Technically, edit command and JSON are separate areas...

 The expected behavior for your given scenario is that VAX@NUS will start up with an empty student book. There are validation checks in place to check that the matriculation number for the appointment belongs to a valid student record containing the corresponding matric number. If the validation fails, VAX@NUS will start up with an empty student book.

Reply:
Yup I don't think it's a contradiction coz the writing commands thru the GUI and editing the JSON are 2 distinct interfaces to interact with the application.

Reply:
Okay, I shall make changes to it! Thank you 👍